### ISSUE TITLE
rules_go 0.15 pins to incompatible versions of go-protobuf and go-genproto

### ISSUE THREAD

**Issue by jmillikin-stripe** [bazelbuild/rules_go#1682]
Context: https://github.com/bazelbuild/rules_go/blob/0.15.0/go/private/repositories.bzl

rules_go pins `@com_github_golang_protobuf` is to v1.1.0, so all `go_protobuf_library()` targets in the repository are forced to use that version of `protoc-gen-go`.  rules_go _also_ pins `@ org_golang_google_genproto`, to a version containing `.pb.go` files generated by `protoc-gen-go` v1.2.0. 

The Go protobuf library explicitly does not guarantee compatibility between generated files from different `protoc-gen-go` plugin versions, and indeed the current situation breaks compilation in common situations:

```
bazel-out/darwin-fastbuild/bin/external/go_googleapis/google/rpc/darwin_amd64_stripped/status_go_proto%/google.golang.org/genproto/googleapis/rpc/status/status.pb.go:22:0: inconsistent definition for type "google.golang.org/genproto/googleapis/rpc/status".Status during import
	struct { Code int32 "protobuf:\"varint,1,opt,name=code,proto3\" json:\"code,omitempty\""; Message string "protobuf:\"bytes,2,opt,name=message,proto3\" json:\"message,omitempty\""; Details []*any.Any "protobuf:\"bytes,3,rep,name=details,proto3\" json:\"details,omitempty\""; XXX_NoUnkeyedLiteral struct {} "json:\"-\""; XXX_unrecognized []byte "json:\"-\""; XXX_sizecache int32 "json:\"-\"" } (in "google.golang.org/genproto/googleapis/rpc/status")
	struct { Code int32 "protobuf:\"varint,1,opt,name=code\" json:\"code,omitempty\""; Message string "protobuf:\"bytes,2,opt,name=message\" json:\"message,omitempty\""; Details []*any.Any "protobuf:\"bytes,3,rep,name=details\" json:\"details,omitempty\""; XXX_NoUnkeyedLiteral struct {} "json:\"-\""; XXX_unrecognized []byte "json:\"-\""; XXX_sizecache int32 "json:\"-\"" } (in "google.golang.org/grpc")
GoCompile: error running subcommand: exit status 2
```

Note that the `",proto3"` struct field tag is present in more places in code generated by `protoc-gen-go` v1.2.

**Comment by jayconrod** [comment]
I'm not sure I follow. `@go_googleapis` is pinned to `master` as of 2018-08-08. `@com_github_golang_protobuf` is pinned to `v1.1.0 as of 2018-08-06. The latest version is `v1.2.0`, released 2018-08-20, after rules_go 0.15 was released.

In any case, the overlay for `@go_googleapis` shouldn't include any rules that reference .pb.go files. That repo is only supposed to include rules that build proto libraries at build time. For `go_library` rules that use .pb.go files, use `@org_golang_google_genproto`.

**Comment by jayconrod** [comment]
As a follow-up to our meeting today, I'll link to [Avoiding conflicts](https://github.com/bazelbuild/rules_go/blob/master/proto/core.rst#avoiding-conflicts) from our proto documentation. The main thing to watch out for is that a binary doesn't mix pre-generated and build-time-generated proto code.
